rust/src/test.rs

```diff
@@ -17,20 +18,125 @@ pub enum TestError {
     Nix(#[from] nix::Error),
 }
 
-/// A single minimal test case
+/// A single minimal test case.
 pub struct TestCase {
     pub name: &'static str,
     pub require_root: bool,
     pub fun: fn(&mut TestContext),
-    pub syscall: Option<ExclFeature>,
+    pub required_features: &'static [FileSystemFeature],
+    #[cfg(any(
+        target_os = "openbsd",
+        target_os = "netbsd",
+        target_os = "freebsd",
+        target_os = "dragonfly",
+        target_os = "macos",
+        target_os = "ios",
+        target_os = "watchos",
+    ))]
+    pub required_file_flags: &'static [FileFlags],
 }
 
 #[distributed_slice]
 pub static TEST_CASES: [TestCase] = [..];
 
-/// Syscalls which are not available on every OS/file system combination.
-#[derive(Debug, PartialEq, Eq, Hash, strum::EnumString, strum::Display, strum::EnumIter)]
+// ANCHOR: file_flags_os
+#[cfg(any(
```

Comment:
I suggest defining `FileFlags` unconditionally, but conditionalizing the definition if each particular flag.  So on Linux for sample it will be an empty enum.  I suspect that will be less confusing overall.

Reply:
That would definitely simplify it! I wasn't sure of how to conditionalize the enum.